pkg/controller/cronjob/metrics.go

```diff
@@ -28,11 +29,50 @@ var schedulingDecisionSkip = prometheus.NewCounterVec(
 	},
 	[]string{namespaceKey, cronNameKey})
 
+var jobSucceeded = prometheus.NewCounterVec(
+	prometheus.CounterOpts{
+		Subsystem: cronjobSubsystem,
+		Name:      "job_succeeded",
+		Help:      "Counter that increments when the cronjob controller detects a child Job has completed with success",
+	}, []string{namespaceKey, cronNameKey})
+
+var jobFailed = prometheus.NewCounterVec(
+	prometheus.CounterOpts{
+		Subsystem: cronjobSubsystem,
+		Name:      "job_failed",
+		Help:      "Counter that increments when the cronjob controller detects a child Job has completed with failure",
+	}, []string{namespaceKey, cronNameKey})
+
+var cronCompletionTimeHistogram = prometheus.NewHistogramVec(
+	prometheus.HistogramOpts{
+		Subsystem: cronjobSubsystem,
+		Name:      "cron_completion_time_histogram_seconds",
+		Help:      "Histogram that observes time in seconds it takes for child Job of CronJob to complete",
+		// this gives us buckets at 1, 2, 4, ... 1024 seconds (17 minutes)
+		Buckets: prometheus.ExponentialBuckets(1, 2, 10),
```

Comment:
This should probably be larger - CronJobs that take a few hours aren't uncommon, and we don't need quite this level of granularity.

Reply:
Deciding on buckets for `job completion time` to cover all cases is inherently quite difficult since those durations are unbounded, and the distribution of the times varies from job to job. 

Another problem is that Prometheus quantile estimation has [some limitations](https://prometheus.io/docs/practices/histograms/#errors-of-quantile-estimation). Setting buckets to large means the long tail end of Job completion times (those that take a long time) would more likely be inaccurate due to larger bucket sizes as the exponent increases.

How would you recommend setting the buckets?

Reply:
The `completion time gauge` metric *does* give an accurate measurement of how long Jobs are taking